modules/auxiliary/scanner/smb/smb_enumshares.rb

```diff
@@ -289,7 +299,16 @@ def run_host(ip)
         smb_login
 
         begin
-          shares = simple.client.net_share_enum_all(ip)
+          # Return all shares if `Share` option has not been set
+          if datastore['Share'].nil?
+            shares = simple.client.net_share_enum_all(ip)
+          else
+            # If the `Share` option has been set attempt to connect to the share to verify it is valid
+            share = "\\\\#{simple.client.default_domain}\\#{datastore['Share']}"
+            simple.client.tree_connect(share)
+            # If the share is valid, then pass it into the shares hash
+            shares = [ { name: datastore['Share'] } ]
```

Comment:
Should we not be checking the return value of this call to ensure we actually connected successfully?

Reply:
After some changes made from @adfoster-r7 's suggestions, it will now revert back to only using the modules existing `tree_connect` .